src/Spryker/Zed/SynchronizationBehavior/Persistence/Propel/Behavior/SynchronizationBehavior.php

```diff
@@ -964,34 +962,71 @@ protected function getLocaleStatement(array $parameters): string
 
     /**
      * @throws \Spryker\Zed\SynchronizationBehavior\Persistence\Propel\Behavior\Exception\MissingAttributeException
-     * @throws \Spryker\Zed\SynchronizationBehavior\Persistence\Propel\Behavior\Exception\InvalidConfigurationException
      *
      * @return string
      */
     protected function getMappings(): string
     {
         $parameters = $this->getParameters();
-        $mappings = [];
-        if (isset($parameters['mappings'])) {
-            if (!isset($parameters['mappings']['value'])) {
-                throw new MissingAttributeException(sprintf(static::ERROR_MISSING_MAPPINGS_PARAMETER, $this->getTable()->getPhpName()));
-            }
-            $mappingsParts = explode(':', $parameters['mappings']['value']);
-            if (count($mappingsParts) !== 2) {
-                throw new InvalidConfigurationException(
-                    sprintf(static::ERROR_INVALID_MAPPINGS_PARAMETER, $this->getTable()->getPhpName())
-                );
-            }
 
-            $mappings = "[
-        [
-            'source' => '{$mappingsParts[0]}',
-            'destination' => '{$mappingsParts[1]}',
-        ],
-    ]";
+        if (!isset($parameters['mappings'])) {
+            return '';
+        }
+
+        if (!isset($parameters['mappings']['value'])) {
+            throw new MissingAttributeException(sprintf(static::ERROR_MISSING_MAPPINGS_PARAMETER, $this->getTable()->getPhpName()));
+        }
+
+        return $this->formatMappingsAsArrayString($parameters['mappings']['value']);
```

Comment:
I don't like `AsArrayString` part, but I don't have better name... Mb we can drop this last part?